### PR TITLE
Create CRUD service for coach's grades

### PR DESCRIPTION
**Pull request by maxim12321**

### CODE REVIEW

File: src/main/java/com/team4/testingsystem/repositories/CoachGradeRepository.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+import javax.transaction.Transactional;
+
+@Repository
+public interface CoachGradeRepository extends CrudRepository<CoachGrade, Long> {
+    Optional<CoachGrade> findByQuestion(Question question);
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/repositories/CoachGradeRepository.java]: A question can be equal in different records.
I think we need to find grades by test if we want to get both of the grades (expected), or by test and question if (for some reason) we are cares about a single mark

File: src/main/java/com/team4/testingsystem/repositories/CoachGradeRepository.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    @Transactional
+    @Modifying
+    @Query(value = "UPDATE CoachGrade cg SET cg.grade = ?2 WHERE cg.id = ?1")
+    int updateGrade(Long id, Integer grade);
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/repositories/CoachGradeRepository.java]: Do we really gonna share grade ids with a client?
I think, for them, it is going to be a new mark for a question in the test.

File: src/main/java/com/team4/testingsystem/services/CoachGradeService.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.team4.testingsystem.services;
+
+import com.team4.testingsystem.entities.CoachGrade;
+import com.team4.testingsystem.entities.Question;
+import com.team4.testingsystem.entities.Test;
+
+public interface CoachGradeService {
+    CoachGrade getGradeById(Long id);
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/services/CoachGradeService.java]: Do we need it?

File: src/main/java/com/team4/testingsystem/services/CoachGradeService.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+    CoachGrade getGradeByQuestion(Question question);
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/services/CoachGradeService.java]: getGradesByTest?

File: src/main/java/com/team4/testingsystem/services/CoachGradeService.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+    void createGrade(Question question, Test test, Integer grade);
+
+    void updateGrade(Long id, Integer grade);
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/services/CoachGradeService.java]: TestId and QuestionId

File: src/main/java/com/team4/testingsystem/entities/CoachGrade.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    private Question question;
+
+    @ManyToOne
+    @JoinColumn(name = "test_id")
+    private Test test;
+
+    public CoachGrade() {
+    }
+
+    public CoachGrade(Integer grade, Question question, Test test) {
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/entities/CoachGrade.java]: Minor:
constructor fields order seems a bit strange.
In a test, for a particular question, we are marking a grade. So I'll propose to use that order.

**Review by AntonMaksimenkau — CHANGES_REQUESTED** [review]

**Review by AntonMaksimenkau — APPROVED** [review]

**Review by Darina007 — APPROVED** [review]

**Pull request merged**